src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -763,13 +765,7 @@ public void execute(Runnable runnable) {
         final Runnable realRunnable = JBossExecutors.classLoaderPreservingTaskUnchecked(runnable);
         int result;
         if (TAIL_LOCK) {
```

Comment:
These conditionals could be replaced by a `NoQueueLock` allowing per-instance configuration at the cost of an additional method invocation.